src/global/utils/auth.tsx

```diff
@@ -16,86 +16,105 @@
  * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-'use client';
+"use client";
 
-import Header from '@/app/components/Header';
-import { DnaLoader } from '@icgc-argo/uikit';
-import Cookies from 'js-cookie';
-import { usePathname, useRouter } from 'next/navigation';
+import Header from "@/app/components/Header";
+import { useAppConfigContext } from "@/app/hooks/AppProvider";
+import createEgoUtils from "@icgc-argo/ego-token-utils";
+import { DnaLoader } from "@icgc-argo/uikit";
+import Cookies from "js-cookie";
+import { usePathname, useRouter } from "next/navigation";
 import {
-	Dispatch,
-	ReactNode,
-	SetStateAction,
-	Suspense,
-	createContext,
-	useContext,
-	useState,
-} from 'react';
-import { EGO_JWT_KEY, LOGIN_NONCE } from '../constants';
+  Dispatch,
+  ReactNode,
+  SetStateAction,
+  Suspense,
+  createContext,
+  useContext,
+  useState,
+} from "react";
+import { EGO_JWT_KEY, LOGIN_NONCE } from "../constants";
 
 type AuthContextValue = {
-	egoJwt: string;
-	setEgoJwt: Dispatch<SetStateAction<string>>;
-	authLoading: boolean;
-	setAuthLoading: Dispatch<SetStateAction<boolean>>;
-	logIn: (newToken: string) => void;
-	logOut: () => void;
+  egoJwt: string;
+  setEgoJwt: Dispatch<SetStateAction<string>>;
+  authLoading: boolean;
+  setAuthLoading: Dispatch<SetStateAction<boolean>>;
+  logIn: (newToken: string) => void;
+  logOut: () => void;
+  permissions: string[];
+  TokenUtils: ReturnType<typeof createEgoUtils>;
 };
 
 const AuthContext = createContext<AuthContextValue>({
-	egoJwt: '',
-	setEgoJwt: () => '',
-	authLoading: false,
-	setAuthLoading: () => false,
-	logIn: () => undefined,
-	logOut: () => undefined,
+  egoJwt: "",
+  setEgoJwt: () => "",
```

Comment:
I don't think this returns a string as type
you're on the right track with the react dispatch generics